openedx/core/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -159,21 +179,28 @@ def test_access_token_fields(self, client_attr):
         self.assertIn('scope', data)
         self.assertIn('token_type', data)
 
-    def test_restricted_access_token_fields(self):
-        response = self._post_request(self.user, self.restricted_dot_app)
-        self.assertEqual(response.status_code, 200)
-        data = json.loads(response.content)
-        self.assertIn('access_token', data)
-        self.assertIn('expires_in', data)
-        self.assertIn('scope', data)
-        self.assertIn('token_type', data)
-
-        # Restricted applications have immediately expired tokens
-        self.assertLess(data['expires_in'], 0)
-
-        # double check that the token stored in the DB is marked as expired
-        access_token = dot_models.AccessToken.objects.get(token=data['access_token'])
-        self.assertTrue(models.RestrictedApplication.verify_access_token_as_expired(access_token))
+    @ddt.data(
+        (False, True),
+        (True, True),
+    )
+    @ddt.unpack
+    def test_restricted_access_token_fields(self, enforce_jwt_scopes_enabled, expiration_expected):
```

Comment:
Rename this so it's clear that it's only for non-JWT based access tokens.  Otherwise, it was initially confusing why the expiration didn't change with `enforce_jwt_scopes_enabled` set to `True`.

Maybe: `test_restricted_non_jwt_access_token`